root/etc/e-smith/templates/etc/ssh/sshd_config/70Restricted2Sftp

```diff
@@ -1,23 +1,57 @@
 {
     # restrict groups to sftp, ssh access is refused
+    return "" if ($sssd{'ShellOverrideStatus'} ne 'enabled');
+
     my $domain = $DomainName;
     my @SftpRestricted = ();
+    my @SftpUnRestricted = ();
+
+    my $admins_group = $admins{'group'} || 'domain admins';
 
-    foreach ( split(',',$sshd{'AllowGroups'} || '')) {
+    foreach ( $admins_group, split(',',$sshd{'AllowGroups'} || '')) {
         my ($group, $sftp) = split(':', $_);
-        if(!$group || !$sftp) {
+        if(!$group) {
             next;
         }
+        if (!$sftp) {
+            $sftp = '';
+        }
+
         $group .= "\@$domain" if ($group !~ '@');
-        push @SftpRestricted, $group if ($sftp eq 'sftp');
+        if ($sftp eq 'sftp') {
+            push @SftpRestricted, $group;
+        } else {
+            # do not force SFTP to groups with SSH permission
+            push @SftpUnRestricted, $group;
+       }
     }
 
-    if (@SftpRestricted) {
-        my $matchGroup = join (",",@SftpRestricted);
-        $OUT .= "Match Group \"$matchGroup\"\n";
+   if ($sshd{'AllowEveryone'} eq 'sftp+ssh') {
+        # Everyone is allowed to SSH & SFTP
+        return  "";
+    } 
+    elsif (@SftpRestricted && ($sshd{'AllowEveryone'} eq 'none')) {
+        # We list non restricted group to SFTP
+        @SftpUnRestricted = join (",",@SftpUnRestricted);
+        # We list restricted group to SFTP
+        @SftpRestricted = join (",",@SftpRestricted);
+
+        $OUT .= "Match Group \"root,@SftpUnRestricted\"\n";
+        $OUT .= "    ForceCommand none\n";
+        $OUT .= "Match Group \"@SftpRestricted\"\n";
         $OUT .= "    ForceCommand internal-sftp\n";
+        $OUT .= "    AllowTCPForwarding no\n";
+        $OUT .= "    X11Forwarding no\n";
+
+    } elsif ($sshd{'AllowEveryone'} eq 'sftp') {
+        # We list non restricted group to SFTP
```

Comment:
Maybe I'm wrong, but this condition is not required: the lines below (48-55) already do the job.

1. The `@SftpRestricted` is not used any more
2. Lines 50-51 grants access to sftp+ssh (SftpUnRestricted)
3. Lines 52-55 restrict access to sftp for those who can access sshd (everyone else)

Consider that "those who can access sshd" are defined as those who have at least sftp access.

Remember that you already excluded the case where Everyone has ssh+sftp access in line 3.

Reply:
??

Reply:
ok got it

Reply:
yep another way...it works